<!-- 
         Velocity smoother
-->
<launch>
    <arg name="ns" default="joy"/>

    <node pkg="nodelet" type="nodelet" name="joy_velocity_smoother_nodelet_manager" args="manager" ns="$(arg ns)"/>

    <!-- Smoother for cmd_vel -->
    <node pkg="nodelet" type="nodelet" name="joy_velocity_smoother" args="load yocs_velocity_smoother/VelocitySmootherNodelet joy_velocity_smoother_nodelet_manager" ns="$(arg ns)">
        <rosparam file="$(find jaime_bringup)/params/navigation/smoother.yaml" command="load"/>
        <remap from="joy_velocity_smoother/smooth_cmd_vel" to="base/cmd_vel"/>

        <!-- Robot velocity feedbacks; use the default base configuration -->
        <remap from="joy_velocity_smoother/odometry" to="/jaime/odom"/>
        <remap from="joy_velocity_smoother/robot_cmd_vel" to="/jaime/mobile_base/commands/velocity"/>
    </node>

    <!-- Smoother for master_cmd_vel-->
    <node pkg="nodelet" type="nodelet" name="joy_master_velocity_smoother" args="load yocs_velocity_smoother/VelocitySmootherNodelet joy_velocity_smoother_nodelet_manager" ns="$(arg ns)">
        <rosparam file="$(find jaime_bringup)/params/navigation/smoother.yaml" command="load"/>
        <remap from="joy_master_velocity_smoother/smooth_cmd_vel" to="base/master_cmd_vel"/>

        <!-- Robot velocity feedbacks; use the default base configuration -->
        <remap from="joy_master_velocity_smoother/odometry" to="/jaime/odom"/>
        <remap from="joy_master_velocity_smoother/robot_cmd_vel" to="/jaime/mobile_base/commands/velocity"/>
    </node>

</launch>
